src/smt.rs

```diff
@@ -0,0 +1,95 @@
+use std::fmt::{Display, Formatter, Result};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SMTVariable {
+    pub name: String,
+}
+
+pub trait SMTSyntaxUnit {
+    fn as_smt(&self) -> String;
+}
+
+impl SMTSyntaxUnit for String {
+    fn as_smt(&self) -> String {
+        self.to_owned()
+    }
+}
+
+pub enum SMTLiteral {
+    Hex(String),
+}
+
+impl SMTLiteral {
+    pub fn hex_zero() -> Self {
```

Comment:
The important aspect of this is that it's a BV constant with 256 bits - maybe we could call this `bv256_zero` or something like that?